pkg/apis/storage/v1/defaults.go

```diff
@@ -31,4 +33,9 @@ func SetDefaults_StorageClass(obj *storagev1.StorageClass) {
 		obj.ReclaimPolicy = new(v1.PersistentVolumeReclaimPolicy)
 		*obj.ReclaimPolicy = v1.PersistentVolumeReclaimDelete
 	}
+
+	if obj.VolumeBindingMode == nil && utilfeature.DefaultFeatureGate.Enabled(features.VolumeScheduling) {
```

Comment:
This doesn't work with upgrades.  Need to discuss with @kubernetes/sig-api-machinery-pr-reviews if it's possible to default new alpha fields.

Reply:
When gate disabled, they should not be persisted or returned from the API, so… not really

Reply:
Discussed with @thockin and @lavalamp and verified that defaults are applied on read to objects that were stored with a new field.  So we can default new fields under a feature gate as long as we also access the fields under the feature gate.

Reply:
Doesn't that mean the default values are serialized back out and returned to API clients even when the gate is disabled? If so, that isn't what we want

Reply:
> So we can default new fields under a feature gate

Ah, you're saying the defaulting itself would be gated. I thought I looked into that at one point and it resulted in weird dependencies, but maybe I'm misremembering

Reply:
Here's a scenario I'm not sure how to handle:

1.  1.9 feature disabled, create new object, nil field not defaulted
2.  1.10 feature is enabled, update object, defaulting will kick in.  How does validation detect the difference between nil field -> defaulting vs nil field -> user set some value.  I want this field to be immutable.

Reply:
> How does validation detect the difference between nil field -> defaulting vs nil field -> user set some value

it doesn't (and shouldn't). the old object decoded from etcd will get the default, get sent to the user, and any update coming from the user without that value will also get the default. from validation's perspective, both the old and new object will have the default value, so the field will not appear to have changed.